DllImportGenerator/DllImportGenerator/Marshalling/SafeHandleMarshaller.cs

```diff
@@ -83,17 +83,36 @@ public IEnumerable<StatementSyntax> Generate(TypePositionInfo info, StubCodeCont
                                                         .WithInitializer(EqualsValueClause(LiteralExpression(SyntaxKind.FalseLiteralExpression))))));
                     
                     }
+
+                    var safeHandleCreationExpression = CastExpression(
+                        info.ManagedType.AsTypeSyntax(),
+                        InvocationExpression(
+                            MemberAccessExpression(
+                                SyntaxKind.SimpleMemberAccessExpression,
+                                ParseTypeName(TypeNames.System_Activator),
+                                IdentifierName("CreateInstance")))
+                        .WithArgumentList(
+                            ArgumentList(
+                                SeparatedList(
+                                    new []{
+                                        Argument(
+                                            TypeOfExpression(
+                                                info.ManagedType.AsTypeSyntax())),
+                                        Argument(
+                                            LiteralExpression(
+                                                SyntaxKind.TrueLiteralExpression))
+                                        .WithNameColon(
+                                            NameColon(
+                                                IdentifierName("nonPublic")))
+                                    }))));
```

Comment:
Can we just use the public parameterless ctor if it has one, and only fall back to Activator if it doesn't?

Reply:
Yes, we should do that. I do recall something about these two approaches being close to the same cost though. Is the primary concern look and feel of the generated code?

Reply:
The reflection approach has much higher fixed footprint. It is a lot cheaper to just call `new MySafeHandle()` where possible.

Reply:
Last I looked, it was small but measurable (e.g. a few percent).  It's also about avoiding dependencies that could otherwise be trimmed away.  And also, yes, the generated code: a small amount of code here in the generator resulting in smaller/better code in tons of P/Invokes.

Reply:
I've updated the codegen to emit a call to the parameterless constructor if it is accessible from the context of the stub.